### PR TITLE
[draftt-69][BE] OAuth2

### PR DESCRIPTION
**Pull request by aarajh**
### Why?
- Improved authentication flow and code related to it.
- Created a new grant-type for external provided tokens.
- Restructured code to separate auth app.
- Added middleware to support email OR username login requests.

All tokens are accepted at /api/auth/token and follow OAuth2 standards.

Need to improve testing.

### CODE REVIEW

File: Server/app/.flake8 [hunk 1]
```diff
@@ -3,4 +3,5 @@ exclude =
     migrations,
     __pycache__,
     manage.py,
-    settings.py
\ No newline at end of file
+    settings.py
+
```
Comment (adil1508) [inline Server/app/.flake8]: wow. dem changes

File: Server/app/core/models.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ class User(AbstractBaseUser, PermissionsMixin):
     fullname = models.CharField(max_length=255, blank=True)
     is_active = models.BooleanField(default=True)
     is_staff = models.BooleanField(default=False)
-    is_verified = models.BooleanField(default=False)
+    is_verified = models.BooleanField(default=True)
```
Comment (adil1508) [inline Server/app/core/models.py]: default is_verified for new Users will be True?

File: Server/app/user/urls.py [hunk 3]
```diff
@@ -5,10 +5,10 @@
 
 """Redirects to the create user or login functions for each request"""
 urlpatterns = [
-    path('create/', views.CreateUserView.as_view(), name='create'),
-    path('login/', views.CreateTokenView.as_view(), name='login'),
+    # path('create/', views.CreateUserView.as_view(), name='create'),
+    # path('login/', views.CreateTokenView.as_view(), name='login'),
     path('account/', views.ManageUserView.as_view(), name='account'),
-    path('logout/', views.LogoutUser.as_view(), name='logout'),
+    # path('logout/', views.LogoutUser.as_view(), name='logout'),
```
Comment (adil1508) [inline Server/app/user/urls.py]: are we not using these paths anymore? 

can remove then?

File: Server/app/core/models.py (hunk 2 above)
Comment (aarajh) [reply Server/app/core/models.py]: I removed some checks for is_verified so i made it to be true. Will change it back when I work on next story and fix the issues.

File: Server/app/user/urls.py (hunk 3 above)
Comment (aarajh) [reply Server/app/user/urls.py]: Need to rewrite it so just disabled it for now...

File: Server/app/.flake8 (hunk 1 above)
Comment (aarajh) [reply Server/app/.flake8]: Blame autoformatting

**Comment by aarajh** [comment]
@adil1508 Pls merge when tested.

**Pull request merged**